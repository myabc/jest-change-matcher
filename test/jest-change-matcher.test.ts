import matcher from '../src/jest-change-matcher'
import jestExpect from 'expect'

jestExpect.extend(matcher)

describe('#toChange', () => {
  it('passes for resolved values that change', () => {
    let a = 0
    expect(() => {
      jestExpect(() => (a += 1)).toChange(() => a)
    }).not.toThrow()
  })

  it(`passes for resolved values that do not change with '.not'`, () => {
    let a = 0
    expect(() => {
      jestExpect(() => (a = a)).not.toChange(() => a)
    }).not.toThrow()
  })

  it('fails for resolved values that do not change', () => {
    let a = 0
    expect(() => {
      jestExpect(() => (a = a)).toChange(() => a)
    }).toThrowErrorMatchingSnapshot()
  })

  it(`fails for resolved values that change with '.not'`, () => {
    let a = 0
    expect(() => {
      jestExpect(() => (a += 1)).not.toChange(() => a)
    }).toThrowErrorMatchingSnapshot()
  })
})
