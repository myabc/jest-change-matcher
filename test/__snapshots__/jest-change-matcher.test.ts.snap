// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toChange fails for 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toChange([22m[32mexpected[39m[2m)[22m

Expected resolved value to change (using ===):
  [32m0[39m
Received:
  [31m0[39m

Difference:

[2mCompared values have no visual difference.[22m"
`;
